### ISSUE TITLE
SRXL2 or SRXLv2 Support

### ISSUE THREAD

**Issue by CloudsCapone** [iNavFlight/inav#5265]
## Current Behavior
<currently, I do not see available protocol to use the newer Spektrum SRXL2 receiver>

## Desired Behavior
<I would love to see option or support>

## Suggested Solution
<If somewhat easily achievable, would be happy with a simple string of CLI commands for the time being.>

## Who does this impact? Who is this for?
<I believe this impacts anyone wishing to use the receiver with iNav>

**Comment by DzikuVx** [comment]
Hi,
So far Spectrum was not very eager to work with us to provide their radio protocols support in INAV. Without Spectrum assistance and probably hardware, INAV developers have zero to none chances to add support.
I suggest contacting Spectrum customer center with a request to support iNAV

**Comment by SpektrumRC** [comment]
Unless we missed an email, I don't recall having been reached out to regarding this, other than some other posts on here from users requesting it.
Our SRXL2 protocol is public and available at https://github.com/SpektrumRC/SRXL2 including both documentation and C library files.
I had previously looked to see if it would have been quick enough to do a merge from betaflight srxl2 implementation into iNav, but there just seemed to be too many other changes/dependencies  that are not yet merged to iNav, and unfortunately I've been too busy to make a new implementation specific to iNav's current state.

**Comment by donsinger1** [comment]
Dzikuvx and SpektrumRC...I asked about this very issue and need in the RC Groups thread on the new SRXL2 receivers, specifically the 6610T and 637T....

I want to use the SRXL2 port on the 6610T to connect to a wing aware flight controller, such as a Matek 411WSE for an iNav build.  I know it has been rolled out on Betaflight (4.1 and above), but wanted to know if there was a list of boards for iNav that had a target that included the SRXL2 protocol.

Andy Kunz replied that I should ask here and "bug the developers" to implement.

From your discussion, to the outsider, it looks like people are pointing fingers at each other, 'everyone is busy', so nothing gets done....It would be great if the development community would embrace Spektrum's new focus on openness and willing to cooperate with 3rd parties; and, I hope that Spektrum does not continue to be snarky as per the reply above and rather respond like 'well, never saw the email but let's work together to get this done'...

Can us users hope to see some progress on this issue? I do not want to have to purchase a FrSky system just to get this capability....

Thanks much for listening...Best, Don

**Comment by MiguelFAlvarez** [comment]
We've still been pretty swamped with stuff at work, but I've taken some personal time to work through adding this. VERY EXPERIMENTAL! Testing/Feedback would be appreciated though.
@donsinger1 Not sure if a 6610T would work, as it handshakes slightly differently than the standalone serial receivers do.

https://github.com/MiguelFAlvarez/inav/releases/tag/2.3.0-SRXL2-TEST

**Comment by donsinger1** [comment]
Hi Miguel:

You are a Prince!!!!

This would be fantastic if it worked on the 6610T…..I have one ready to test and a few blank Matek boards to try on…since I am doing a build right now, I can incorporate this with the FrSky RX with SBUS I had planned to use and test servos, esc, etc…will do the RX’s in parallel.

Can’t wait to play with this tonight.

Cordially,
Don Singer
KF7DS

**Comment by donsinger1** [comment]
Hi Miguel:

I also have a few 4650s so will try those as well.

Don

**Comment by Romulusg** [comment]
Has any of this been included in 2.4? Or do I need to use 2.3.0? Thanks for the work on this.

**Comment by MiguelFAlvarez** [comment]
> Has any of this been included in 2.4? Or do I need to use 2.3.0? Thanks for the work on this.

I haven't received any feedback from anybody, and the only testing that has been done is from my own personal use (no issues so far), so I haven't made a merge request yet. As of right now, it's only available on 2.3 from the custom builds on my page. Haven't compiled for 2.4 yet since its still in the release candidate phase.

**Comment by Romulusg** [comment]
Thanks for the info. I will test it next weekend when I can. What type of feedback would you like? 

Todd

**Comment by MiguelFAlvarez** [comment]
@Romulusg Mainly just set it up and make sure it works properly (channel data, telemetry, rssi, failsafe detection) and doesn't cause any issues to otherwise normal operation like FC freezing/crashing. I only have F4 FC's so that's all I've tested personally. No idea if it works for F7's and F3's dont have enough flash to support the feature, so it's not implemented for those. Even if you test on F4's just be cautious and test everything on the bench. While it's not likely, its possible that some combination of features might cause an issue that I'm not seeing.

**Comment by Romulusg** [comment]
I wasnt able to test this weekend I couldnt get access to my srxl2 receiver.  I ordered another one and will update as I test.

**Comment by Romulusg** [comment]
Hi Miguel,

I have loaded the hex file and copied the cli info.  So far on my matekf411 wing FC everything appears working.  controls function as expected and modes switch as well.

Auto launch works on the bench with a prop off and auto level is working as well 

I havent tried RTH as it is on the bench.

So far it seems good.  

I will try a flight when the windy chinook wind in calgary drops hopefully this weekend.

**Comment by Romulusg** [comment]
I installed the spm4650 spektrum rx and put it on TX1

one side note in the CLI the receiver is SRXL2 and it shows as valid but it doesnt appear in the receiver drop down.  My guess it the array for the list range needs to be expanded by a row as it appears last in the list.

**Comment by SpektrumRC** [comment]
@Romulusg Thanks for the feedback. Look forward to hearing more.
In regards to the configurator receiver drop down, it wont appear there until the configurator is also updated. For now, the cli is the only option to set this. I will make the appropriate configurator changes as well once a merge request for the protocol is submitted.

**Comment by Romulusg** [comment]
Ok additional testing this evening.  I was able to enter the betaflight menu on the transmitter telemetry screen and change vtx settings as expected. another thumbs up.

**Comment by SpektrumRC** [comment]
@Romulusg btw, I compiled these changes onto the 2.4 release last night, but did not get a chance to test. Will try to test tonight and if it looks good on the bench, I'll upload that

**Comment by trickapm** [comment]
I can test a 4650 receiver with a matek f722 if i can get a link to the 2.4.0 builds?

**Comment by MiguelFAlvarez** [comment]
@trickapm https://github.com/MiguelFAlvarez/inav/releases

**Comment by trickapm** [comment]
On the bench all looks well, didn't notice any issues. But i may not really know what to look for either ;-). I already bound it to my tx using the button on the pcb so didn't use the bind_rx command.

**Comment by AntDX316** [comment]
need SRXL2 support on 2.4.1

**Comment by Romulusg** [comment]
@AntDX316 above there is a link to 2.4.0 there has not been an merge yet as far as @MiguelFAlvarez has mentioned in this page.

**Comment by AntDX316** [comment]
> @AntDX316 above there is a link to 2.4.0 there has not been an merge yet as far as @MiguelFAlvarez has mentioned in this page.

Do I have to reflash to the firmwares underneath the post or can I just drop the CLI in 2.4.1??

**Comment by Romulusg** [comment]
@MiguelFAlvarez can correct me if I’m incorrect but I believe the code needed for the flight controller hex files is in the link above. You would need to flash to that version as the cli commands are reliant on the code in the background.

**Comment by AntDX316** [comment]
> @MiguelFAlvarez can correct me if I’m incorrect but I believe the code needed for the flight controller hex files is in the link above. You would need to flash to that version as the cli commands are reliant on the code in the background.

There are a lot of files that are in the hex.. I'm sure it wasn't done individually?  It much be an easy process to inject?  Now, can we get the steps required to inject the code for it to work?  Just incase the maker stops supporting for some reason?

**Comment by Romulusg** [comment]
@MiguelFAlvarez i flew my inav setup wing with the srxl2 receiver and everything worked seamlessly. How does this get merged?

**Comment by donsinger1** [comment]
That is great…been way too busy the last three months to contribute, but glad to see there is substantial progress. Thanks Miguel and everyone else.

Stay safe.

Best
Don Singer
Portland, OR

**Comment by MiguelFAlvarez** [comment]
I'll start doing pull requests for the necessary changes

**Comment by MiguelFAlvarez** [comment]
First pull request for uart idle callback support https://github.com/iNavFlight/inav/pull/5540 which is required for SRXL2 to work.
Once that's merged, I will make a pull request for the SRXL2 support.

**Comment by AntDX316** [comment]
> First pull request for uart idle callback support #5540 which is required for SRXL2 to work.
> Once that's merged, I will make a pull request for the SRXL2 support.

Are you going to update it to 2.4.1?

**Comment by MiguelFAlvarez** [comment]
> > First pull request for uart idle callback support #5540 which is required for SRXL2 to work.
> > Once that's merged, I will make a pull request for the SRXL2 support.
> 
> Are you going to update it to 2.4.1?

I'm making a pull request for the master branch. From there it should start getting included in upcoming releases. As for including it in my personal builds, I've only been compiling my changes on the latest release, and there is no 2.4.1 release yet unless I'm missing something, only 2.4.0. https://github.com/iNavFlight/inav/releases

**Comment by AntDX316** [comment]
> > > First pull request for uart idle callback support #5540 which is required for SRXL2 to work.
> > > Once that's merged, I will make a pull request for the SRXL2 support.
> > 
> > 
> > Are you going to update it to 2.4.1?
> 
> I'm making a pull request for the master branch. From there it should start getting included in upcoming releases. As for including it in my personal builds, I've only been compiling my changes on the latest release, and there is no 2.4.1 release yet unless I'm missing something, only 2.4.0. https://github.com/iNavFlight/inav/releases

What do you mean a pull request from the master branch?  How long could that possibly take and is it even worth upgrading to 2.4.1 from 2.4.0? : )

**Comment by MiguelFAlvarez** [comment]
@AntDX316 I'm not sure how long it'll take. It's up to the inav devs when they have time to review/approve the code changes, and when they plan on making the next official release.
I'm not sure where you're getting 2.4.1. There is no 2.4.1 inav release yet.

**Comment by AntDX316** [comment]
> @AntDX316 I'm not sure how long it'll take. It's up to the inav devs when they have time to review/approve the code changes, and when they plan on making the next official release.
> I'm not sure where you're getting 2.4.1. There is no 2.4.1 inav release yet.

2.4.1 has been out
https://github.com/iNavFlight/inav-configurator/releases

**Comment by MiguelFAlvarez** [comment]
That is the configurator, not the firmware.

**Comment by AntDX316** [comment]
> That is the configurator, not the firmware.

so 2.4.0 that you put the SRXL2 on is the latest? : )

**Comment by MiguelFAlvarez** [comment]
> > That is the configurator, not the firmware.
> 
> so 2.4.0 that you put the SRXL2 on is the latest? : )

yup!

**Comment by AntDX316** [comment]
> > > That is the configurator, not the firmware.
> > 
> > 
> > so 2.4.0 that you put the SRXL2 on is the latest? : )
> 
> yup!

I'm trying to flash with load firmware [local] w/ the fix but it doesn't work? : (

**Comment by AntDX316** [comment]
> > > That is the configurator, not the firmware.
> > 
> > 
> > so 2.4.0 that you put the SRXL2 on is the latest? : )
> 
> yup!

Can you push it through so we can have SRXL2 native support with 2.5 so I can load firmware [Online] and it work?

**Comment by AntDX316** [comment]
> > > That is the configurator, not the firmware.
> > 
> > 
> > so 2.4.0 that you put the SRXL2 on is the latest? : )
> 
> yup!

What do you set in receiver protocol? 2048??

**Comment by AntDX316** [comment]
> > > That is the configurator, not the firmware.
> > 
> > 
> > so 2.4.0 that you put the SRXL2 on is the latest? : )
> 
> yup!

Works perfectly!

**Comment by Romulusg** [comment]
With the release of 2.5 has the merge of srxl2 been added?  I dont see it in the release notes. Or is this fake
https://github.com/iNavFlight/inav/wiki/2.5.0-Release-Notes
@MiguelFAlvarez

**Comment by MiguelFAlvarez** [comment]
Not yet, the devs requested some changes to the code that I've just not had a chance to address

**Comment by AntDX316** [comment]
> Not yet, the devs requested some changes to the code that I've just not had a chance to address

Please have it where there is native SRXL2 support w/ no workaround.

**Comment by Romulusg** [comment]
Ok thanks for the update.

**Comment by rickgode** [comment]
Hi guys!  I flashed the hex above to my matek 405 mini and using a SPM4650 Rx and it seems to work perfectly on the bench.

**Comment by donsinger1** [comment]
What version…2.4 or 2.5?

Don

**Comment by rickgode** [comment]
Edit:  I didn't see his 2.4.0 version.  I just flashed that hex and everything still works.

Mine says

version
INAV/MATEKF405 2.3.0 Jan 15 2020 / 22:30:44 (e5b5aad2c)
GCC-9.2.1 20191025 (release) [ARM/arm-9-branch revision 277599]

Ricky

**Comment by tourerjim** [comment]
For Inav 2.5 could someone guide for the SRXL2 file so I can use my Sppedybee F4 with the SPM4650 receiver.

**Comment by AntDX316** [comment]
What takes so long to get native SRXL2 support?  Betaflight already has it.

**Comment by tourerjim** [comment]
> For Inav 2.5 could someone guide for the SRXL2 file so I can use my Sppedybee F4 with the SPM4650 receiver.

I've now got my quad built and have Inav 2.5 loaded on the Speedybee FC but without the SRXL2 I can't complete it and just having retired would loved to get out and try it, I copped the above Cli SRXL2 file but doesn't work, Help

**Comment by MiguelFAlvarez** [comment]
@tourerjim inav doesn't officially support it yet. I'm working on that. You need a custom build from here in the meantime https://github.com/MiguelFAlvarez/inav/releases

**Comment by Romulusg** [comment]
> 
> 
> @tourerjim inav doesn't officially support it yet. I'm working on that. You need a custom build from here in the meantime https://github.com/MiguelFAlvarez/inav/releases

@MiguelFAlvarez does that link and hex files then work with version 2.5? 

Thanks

**Comment by MiguelFAlvarez** [comment]
In case anyone is keeping track of the state of implementation, PR #5757 has been approved (uart driver dependencies for SRXL2). Now the actual SRXL2 merge request has been submitted as PR #5791

**Comment by rickgode** [comment]
Sweet!  I was able to maiden my 2.4 inav with SRXL2 support for the SPM4650 Rx on a Matek 405 mini and it works great.  rssi works along with the telemetry TEXT GEN.

Here is some flight cam footage https://youtu.be/APn3u8ktu9E

Testing out auto launch https://youtu.be/ftnLqZ7Sk5c

**Comment by tourerjim** [comment]
I've paste the SRXL2 file into Cli but get these Errors anyone know what could I be doing wrong.
![IMG_20200530_081429](https://user-images.githubusercontent.com/65942690/83322410-46abc600-a24f-11ea-92b2-d50b30b8e28d.jpg)

**Comment by MiguelFAlvarez** [comment]
> I've paste the SRXL2 file into Cli but get these Errors anyone know what could I be doing wrong.
> ![IMG_20200530_081429](https://user-images.githubusercontent.com/65942690/83322410-46abc600-a24f-11ea-92b2-d50b30b8e28d.jpg)

You need to install the custom firmware I linked above. That's why it thinks those parameters are invalid

**Comment by tourerjim** [comment]
> @tourerjim inav doesn't officially support it yet. I'm working on that. You need a custom build from here in the meantime https://github.com/MiguelFAlvarez/inav/releases

Really appreciate your help, now got my SPM4650 working with Inav so now going to go through and set everything up,Thanks

**Comment by tourerjim** [comment]
> > @tourerjim inav doesn't officially support it yet. I'm working on that. You need a custom build from here in the meantime https://github.com/MiguelFAlvarez/inav/releases
> 
> @MiguelFAlvarez does that link and hex files then work with version 2.5?
> 
> Thanks

Yes, I've got mine working.

MiguelFAlvarez
Really appreciate your help, now got my SPM4650 working with Inav so now going to go through and set everything up,Thanks

**Comment by AntDX316** [comment]
> In case anyone is keeping track of the state of implementation, PR #5757 has been approved (uart driver dependencies for SRXL2). Now the actual SRXL2 merge request has been submitted as PR #5791

let us know when it's fully implemented so we can download it w/o a workaround : )

**Comment by donsinger1** [comment]
Thanks much Miguel…been patient and appreciate your work on this…

Don

**Comment by Hineswoodworks** [comment]
Im hoping someone can help me out. I'm a little stumped. I'm flying a FT Mini Arrow with a Matek F405-Wing controller. I'm trying to set up a SPM4650 receiver (didn't realize it was SRXLv2 when I ordered it). I'm using a DXe radio (not sure if that makes any difference).

I have it wired to the tx2 pin. 
I flashed the hex from Miguel (inav_2.4.0_MATEKF405SE.hex under assets right?)
I copied the CLI settings and had no errors.
Bound my receiver / radio

When I go to the receiver tab in iNav none of the channels are responding to movement on the radio.

Did I miss something? Did I use the right hex file? Where do I start my trouble shooting.

Any help would be appreciated.

Thanks

**Comment by MiguelFAlvarez** [comment]
@Hineswoodworks you need to assign the uart port in the ports tab before you do the CLI step.

**Comment by Hineswoodworks** [comment]
I had the wing flying on a different receiver before that I had borrowed from my quad. It was wired to rx2 and the new one is on tx2. UART2 is still set for serial RX. Do I need to do something different with how its set?

**Comment by tourerjim** [comment]
In the Inav receiver tab did you select Spek 'TAER'

**Comment by Hineswoodworks** [comment]
Yes "TAER" is selected in the receiver tab.

**Comment by tourerjim** [comment]
> @MiguelFAlvarez does that link and hex files then work with version 2.5?
> 
> Thanks

SRXL2 does works on 2.5 but can't save PID's poss due to FC hex is only 2.4

**Comment by Hineswoodworks** [comment]
Ok well found a solution. I moved it from UART2 to UART3 and it worked as expected. I'm not overly happy that solution only because I don't like that I don't know why UART2 didn't work.

**Comment by tourerjim** [comment]
> Ok well found a solution. I moved it from UART2 to UART3 and it worked as expected. I'm not overly happy that solution only because I don't like that I don't know why UART2 didn't work.

Strange..! had my Speedybee F4 on Uart2 TX2 for SPM4650 and worked, changed it to Uart3 as Uart2 was nearer for my smartaudio

**Comment by MiguelFAlvarez** [comment]
I meant to answer this earlier but got caught up with some other tasks. What I've noticed is some FC's have TX pins that have hardware inverter needed for S.Port telemetry. Unfortunately, these inverters prevent bidirectional communication so they are incompatible with SRXL2. The solution is usually just use a different UART.

**Comment by tourerjim** [comment]
> I meant to answer this earlier but got caught up with some other tasks. What I've noticed is some FC's have TX pins that have hardware inverter needed for S.Port telemetry. Unfortunately, these inverters prevent bidirectional communication so they are incompatible with SRXL2. The solution is usually just use a different UART.

Aww that's helpful to know as I'm going to put Inav on another FC as well.,

**Comment by tourerjim** [comment]
Just got Inav update to 2.5 and still no SRXL2 and now SpeedyBee F4 Hex file is missing from the FC list.
Does that mean I just copy 2.4's SRXL2 into Cli as I cant find it for 2.5

**Comment by donsinger1** [comment]
I just noticed this as well…I guess just have to be patient.

Don

**Comment by MiguelFAlvarez** [comment]
The pull request has been approved today, and should make it into 2.6
In the meantime, I'll compile targets for use with 2.5. Standby

**Comment by MiguelFAlvarez** [comment]
I've posted 2.5.0 releases now https://github.com/MiguelFAlvarez/inav/releases
This time I've included F3 targets as its quicker for me to compile all targets than try to figure out which are F3's and remove them from the compile list. This does not mean they will work with SRXL2 though.

**Comment by Romulusg** [comment]
Miguel,

Thank you for tour time and effort on this integration. I, and I know others appreciate it. 

Have a nice evening. 

Todd

**Comment by donsinger1** [comment]
Yes, thanks Miguel, it is very appreciated

Don

**Comment by tourerjim** [comment]
> Yes, thanks Miguel, it is very appreciated Don From: Romulusg <notifications@github.com> Sent: Monday, June 8, 2020 6:11 PM To: iNavFlight/inav <inav@noreply.github.com> Cc: donsinger1 <donsinger@outlook.com>; Mention <mention@noreply.github.com> Subject: Re: [iNavFlight/inav] SRXL2 or SRXLv2 Support (#5265) Miguel, Thank you for tour time and effort on this integration. I, and I know others appreciate it. Have a nice evening. Todd
> On Jun 8, 2020, at 3:15 PM, MiguelFAlvarez ***@***.***> wrote: ﻿ I've posted 2.5.0 releases now https://github.com/MiguelFAlvarez/inav/releases This time I've included F3 targets as its quicker for me to compile all targets than try to figure out which are F3's and remove them from the compile list. This does not mean they will work with SRXL2 though. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or unsubscribe.
> — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FiNavFlight%2Finav%2Fissues%2F5265%23issuecomment-640968506&data=02%7C01%7C%7C19cd78f8dcbc48ae7ac808d80c11ecc6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637272618402450362&sdata=V%2BJOMS0TS3AnoeOBebysVdlttlwMETnC3tONx%2FB9%2BXw%3D&reserved=0>, or unsubscribe<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAH64I3SLO4WZONZSUKX6LC3RVWDY5ANCNFSM4JYY6V4A&data=02%7C01%7C%7C19cd78f8dcbc48ae7ac808d80c11ecc6%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637272618402455343&sdata=lUeQo4ViglXNTk3pn0GF0AlxY%2FDbPTga9rJTnYOmlGA%3D&reserved=0>.

This is appreciated, thanks and soon as 2.6 is here im moving all my quads over.

**Comment by AntDX316** [comment]
> The pull request has been approved today, and should make it into 2.6
> In the meantime, I'll compile targets for use with 2.5. Standby

So at 2.6 we will have full SRXL2 support? : )

**Comment by tourerjim** [comment]
> > The pull request has been approved today, and should make it into 2.6
> > In the meantime, I'll compile targets for use with 2.5. Standby
> 
> So at 2.6 we will have full SRXL2 support? : )

Brilliant news, thank you

**Comment by Momike-6059** [comment]
I am just finishing my first build and wanted to use Inav with SPM4650 receiver.  I am not a programmer so am hoping someone can help me understand what needs done to make it work.  Any help is appreciated.

**Comment by tourerjim** [comment]
> I am just finishing my first build and wanted to use Inav with SPM4650 receiver. I am not a programmer so am hoping someone can help me understand what needs done to make it work. Any help is appreciated.

You need the Inav Hex file that relates to your FC https://github.com/MiguelFAlvarez/inav/releases and then you need to copy the spm4650 file and and save to Cli

**Comment by Momike-6059** [comment]
Thanks, but I cannot open the hex file.  What am I supposed to do with it?  As I said, i am not used to dealing with code/programming.

**Comment by tourerjim** [comment]
Download the hex file you need and go into Inav firmware updater and load hex file from pc, once your FC has updated go back to github and copy the Cli file and paste into Inav Cli, then in the 'Receiver Tab select SRXL2 and you should be good to go

**Comment by Momike-6059** [comment]
One last stupid question, how do I go into the nav firmware updater?

**Comment by tourerjim** [comment]
Your nede to put your FC into bootloader then on left of Inav startup screen your see firmwarw updater, there's a few videos on how to load firmware from pc and putting your FC into bootloader

**Comment by Momike-6059** [comment]
It worked! Thanks so much!

**Comment by aur3li3n** [comment]
is that all you have yo do to get srxl2 to work on inav because i followed the instructions and it still isnt working, the reciver tab isnt moving when given input.

**Comment by Momike-6059** [comment]
So I downloaded the patch, installed it with full erase, then turned off
full erase and reloaded inaccurate, then it worked.

**Comment by Momike-6059** [comment]
That’s inav not in accurate, stupid auto correct.

**Comment by tourerjim** [comment]
Just downloaded Inav 2.6 configurator and flashed hex file but can't see the SRXL2 serial in the receiver tab, is there a link to the hex file that contains the SRXL2

**Comment by stronnag** [comment]
> Just downloaded Inav 2.6 configurator and flashed hex file but can't see the SRXL2 serial in the receiver tab, is there a link to the hex file that contains the SRXL2

The PR implementing SRXL2 (#5791) has not been merged into master and currently has a number of (trivial) conflicts that prevent it from being merged for 2.6. We have also not heard from the developer for a while.

**Comment by rickgode** [comment]
Just throwing this out there since it has now closed, I just updated my MATEK405 board from MiguelFAlvarez's 2.5 release to the official inav 2.6 release following these steps.

1.  Download and install the new configurator
2.  Save to a file the current diff all from the CLI.
3.  Upgrade to INAV 2.6 using the Full Erase option in the configurator.
4.  Upload your OSD font of choice from the OSD tab.
5.  Go the CLI again and paste the contents on the file you previously create and write save , press ENTER and done!
6.  You should be ready, explore new 2.6 features and enjoy!

Everything seems to be working great!  Thanks everyone for your efforts!

Ricky
